connector_search_engine: improve state update wiz

simahawk opened the pull request:
Allow to schedule jobs right away instead of waiting for the cron to pass by.

bealdav reviewed (approved):
Nice feature, thanks a lot.

LGTM on the code side

simahawk commented on connector_search_engine/wizards/se_binding_state_updater.py:
```diff
@@ -14,7 +14,12 @@ class SeBindingStateUpdater(models.TransientModel):
+        bindings.write({"state": self.state})
+        if self.do_it_now:
+            domain = [("id", "in", bindings.ids)]
+            bindings.index_id.generate_batch_recompute_per_index(domain=domain)
```
I just drafted this quickly and in fact is not working because the domain is meant to be for indexes. I'm working on a proper implementation.

simahawk committed:
connector_search_engine: allow recompute specific bindings

_jobify_batch_recompute and batch_recompute now accept a binding_ids param

to force the records to be computed.

cyrilmanuel commented:
Hi @lmignon  is it good for you ?

lmignon commented on connector_search_engine/models/se_index.py:
```diff
@@ -216,12 +216,14 @@ def _jobify_batch_sync(self, force_export: bool = False) -> None:
 
-    def _jobify_batch_recompute(self, force_export: bool = False) -> None:
+    def _jobify_batch_recompute(
+        self, force_export: bool = False, binding_ids: list | None = None
```
```suggestion
        self, force_export: bool = False, binding_ids: list[int] | None = None
```

lmignon commented on connector_search_engine/models/se_index.py:
```diff
@@ -256,11 +258,16 @@ def _get_domain_for_recomputing_binding(self, force_export: bool = False) -> lis
 
-    def batch_recompute(self, force_export: bool = False) -> None:
+    def batch_recompute(
+        self, force_export: bool = False, binding_ids: list | None = None
```
```suggestion
        self, force_export: bool = False, binding_ids: list[int] | None = None
```

sbidoul replied:
What do you think about doing another method? (`batch_recompute_bindings_ids`?), and factoring out the common part. Less if's is good, in general.

lmignon reviewed (commented):
@simahawk 2 little comments but LGTM. Thank you for this improvement.

sbidoul commented on connector_search_engine/models/se_index.py:
```diff
@@ -216,12 +216,14 @@ def _jobify_batch_sync(self, force_export: bool = False) -> None:
         self.with_delay(
             description=description, identity_key=identity_exact
-        ).batch_recompute(force_export)
+        ).batch_recompute(force_export, binding_ids=binding_ids)
```
Strictly speaking this job description is now inaccurate because it is now a partial recompute? So maybe this method should be split too?

lmignon commented:
/ocabot merge patch

cyrilmanuel commented:
2026-02-09 10:06:17,194 337 ERROR odoo odoo.addons.connector_search_engine.tests.test_all: FAIL: TestBindingIndex.test_life_cycle
Traceback (most recent call last):
  File "/__w/search-engine/search-engine/connector_search_engine/tests/test_all.py", line 250, in test_life_cycle
    self.assertEqual(self.partner_binding.state, "to_export")
AssertionError: 'recomputing' != 'to_export'
- recomputing
+ to_export

seems that something wrong on this tests !

cyrilmanuel committed:
fixup! connector_search_engine: allow recompute specific bindings

cyrilmanuel commented:
hi @lmignon, i update the code with your comment and i rebase the branch. could you redo the merge please :) 
thanks a lot

lmignon commented:
/ocabot merge patch
